tests/functional/s3_acl_metadata.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+AWS="aws --endpoint-url http://localhost:5000 --no-verify-ssl"
+
+BUCKET=bucket-$RANDOM
+
+# AWS="aws --profil amazon"
+# BUCKET=openio.michael
+
+echo "Bucket name: $BUCKET"
+
+set -x
+set -e
+
+${AWS} s3api create-bucket --bucket ${BUCKET}
+
+${AWS} s3api put-object --bucket ${BUCKET} --key small --body /etc/passwd --acl public-read-write --metadata key1=val1,key2=val2
+
+${AWS} s3api copy-object --bucket ${BUCKET} --copy-source ${BUCKET}/small --key copy
+
+# check metadata of copied object
+data=$(${AWS} s3api head-object --bucket ${BUCKET} --key copy)
+echo "$data" | grep key1
+
+# check ACL of copied object: it should be reset !
+data=$(${AWS} s3api get-object-acl --bucket ${BUCKET} --key copy)
+if [ $(echo "$data" | grep -c Grantee) -ne 1 ]; then
+    echo "Invalid data"
+    exit 1
+fi
+
+### METADATA
+
+${AWS} s3api copy-object --bucket ${BUCKET} --copy-source ${BUCKET}/small --key copy --metadata key3=val3,key4=val4
+
+# since --metadata-directive REPLACE was not specified, metadata are ignored
+data=$(${AWS} s3api head-object --bucket ${BUCKET} --key copy)
+echo "$data" | grep key1
+
+# FIXME(mbo) ISSUE HERE
+echo "$data" | grep key3 && echo "new metadata should be ignored"
```

Comment:
Don't forget to make this fail.

Reply:
Nope, I've added a reference to this known issue on Swift3 side (https://bugs.launchpad.net/swift3/+bug/1433875)